Ofir.yefet/add api catalog documentation (#21899)
DataDog/documentation: content/en/api_catalog/endpoint_discovery.md

---

> title: Endpoint Discovery from APM

Reviewer: ```suggestion
title: Discovering Endpoints from APM
```
^ Just an idea to match the title structure of our other pages.

kind: documentation
is_beta: true
further_reading:
- link: "/tracing/api_catalog/"
  tag: "Documentation"
  text: "Datadog API Catalog"
aliases:
    - /tracing/api_catalog/endpoint_discovery/
---

## Overview

> If a service has a supported tracer installed, the API Catalog will automatically be filled by all endpoints discovered from this service. 

Reviewer: ```suggestion
If a service has a supported tracer installed, the API Catalog is automatically populated with all endpoints discovered from this service. 
```

> In order to check compatibility, you can visit the ‘troubleshoot’ page on the app by pressing the ‘learn more’ button

Reviewer: ```suggestion
To check for compatibility issues, click **Learn More** in the app, then select **Troubleshoot**.
```

{{< img src="tracing/api_catalog/api-catalog-discovery-learn-more.png" alt="ALT TEXT" style="width:30%;text-align: left;" >}}

> ## Providing endpoint path manually

Reviewer: ```suggestion
## Providing endpoint paths manually
```

> Add the `datadog.api_catalog.route` tag to force endpoint discovery by the API Catalog.

Reviewer: ```suggestion
Add the `datadog.api_catalog.route` tag to a service to force endpoint discovery by the API Catalog.
```
^ You would add this tag to your service's code, right?

This tag should contain the matched route, that is, the path template in the format used by the respective server framework.

> Note: API Catalog will still filter out spans that do not contain the following tags:
> * `http.method`
> * `http.status_code`

Reviewer: ```suggestion
<div class="alert alert-info">API Catalog filters out spans that do not contain the following tags:<ul>
<li> <code>http.method</code>
<li> <code>http.status_code</code></ul>
</div>
```
This is how we can make alert boxes in the docs that look like this:
![image](https://github.com/DataDog/documentation/assets/84536271/68d70773-c1cc-4ffa-9e3b-fafd1c562727)

## Example

> Example of adding a custom tag for each span in Go and Ruby:

Reviewer: ```suggestion
The following examples demonstrate adding a custom tag for each span in Go and Ruby:
```

**Go**\
{{< code-block lang="go" disable_copy="true" >}}
span.SetTag("datadog.api_catalog.route", "/products/:id")
{{< /code-block >}}
**Ruby**\
{{< code-block lang="ruby" disable_copy="true" >}}
Datadog::Tracing.active_trace.set_tag('datadog.api_catalog.route', '/products/:id')
{{< /code-block >}}

> Once you set this tag, you should expect to see this tag in spans:

Reviewer: ```suggestion
After you set this tag, you should see this tag in spans:
```

{{< img src="tracing/api_catalog/api-catalog-discovery-span.png" alt="ALT TEXT" style="width:100%;" >}}

> Which will lead to the creation of the following endpoint in the API Catalog:

Reviewer: ```suggestion
This leads to the endpoint displaying in the API Catalog:
```
^ This doesn't create an endpoint--it just shows in the catalog, right?
